tests/test_adb_device.py

```diff
@@ -571,6 +749,53 @@ def test_push_file(self):
             self.device.push('TEST_FILE', '/data', mtime=mtime)
             self.assertEqual(expected_bulk_write, self.device._transport._bulk_write)
 
+    def test_push_issue113(self):
+        # pytest tests/test_adb_device.py::TestAdbDevice::test_push_issue113 --log-cli-level=INFO
```

Comment:
@7homasSutter this command will run this unit test alone. The `filedata` isn't right, but the rest of the test corresponds to what you posted in your issue. It looks like the Android emulator never sent a response.

Reply:
Thanks a lot!